src/ansys/mapdl/core/parameters.py

```diff
@@ -326,6 +330,13 @@ def __repr__(self):
 
     def __getitem__(self, key):
         """Return a parameter"""
+        if self._mapdl._store_commands:
+            raise MapdlRuntimeError(
+                "Cannot use `mapdl.parameters` to retrieve parameters when in "
+                "`non_interactive` mode. "
+                "Exit non_interactive mode before using this method."
+            )
+
         if not isinstance(key, str):
             raise TypeError("Parameter name must be a string")
```

Comment:
Message strings should have concluding punctuation. I didn't place a period here only because I didn't want to possibly break a test.

Reply:
Very thoughtful!